joblib/_memmapping_reducer.py

```diff
@@ -57,6 +61,75 @@
 FOLDER_PERMISSIONS = stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR
 FILE_PERMISSIONS = stat.S_IRUSR | stat.S_IWUSR
 
+# Set referencing the filenames of temporary memmaps created by joblib to speed
+# up data communication between workers. This object need not be specific to a
+# Parallel object unless we decide to support the case of calling several
+# Parallel object run in different threads of a same process.
+JOBLIB_MMAPS = set()
+
+JOBLIB_MMAPS_FINALIZERS = []
+
+
+def add_maybe_unlink_finalizer(memmap):
+    util.debug(
+        "[FINALIZER ADD] adding finalizer to {} (id {}, filename {}, pid  {} )"
+        "".format(type(memmap), id(memmap), os.path.basename(memmap.filename),
+                  os.getpid()))
+    filename = memmap.filename
+
+    def _maybe_unlink(wr):
+        # A weakref callback is constrained to be a function with one
+        # argument that is the weakref itself, while the
+        # resource_tracker.maybe_unlink needs the filename and the type of
+        # the resource to unlink. To workaround this limitation, we create
+        # a nested function whose closure will contain the arguments of
+        # resource_tracker.maybe_unlink.
+        from .externals.loky.backend.resource_tracker import _resource_tracker
+        util.debug(
+            "[FINALIZER CALL] object mapping to {} about to be deleted,"
+            " decrementing the refcount of the file (pid: {})".format(
+                os.path.basename(filename), os.getpid()))
+        _resource_tracker.maybe_unlink(filename, "file_plus_plus")
+
+    if sys.version_info[0] == 2:
+        w = weakref.ref(memmap, _maybe_unlink)
+        JOBLIB_MMAPS_FINALIZERS.append(w)
+    else:
+        weakref.finalize(memmap, _maybe_unlink, (None,))
+
+
+def file_plus_plus_unlink(filename):
```

Comment:
Sorry for the back and forth, but for the sake of the code simplicity I would be in favor not implementing this and instead try to best effort delete the temp folders at the end of each parallel calls (in terminate) and otherwise delegate the deletion to atexit with the ResourceTracker as final backup in case of hard crash of the main process